### PR TITLE
D8CORE-2444: Changing the anchor to a div for the landing of the anchor

### PR DESCRIPTION
**Pull request by jenbreese**
# READY FOR REVIEW
- I'd like to get some commentary on how to make this better or a different approach.  - DONE
- Removed all the JS changes and move to PHP.
- Received feedback from Clare from SOAP.
- Needs a review by @cjwest 

# Summary
- Changing the skip link target from an anchor to a div

# Needed By (Date)
- 4/12

# Urgency
- Med

# Steps to Test

1. Checkout this branch and pull in this change.
2. Clear the caches.
3. Got to a page on your site in an incognito mode. 
4. Press Tab first to get the Skip to main link.
5. Press Enter when when the Skip Link has focus.
6. Notice you jump to a div.
7. If you press "Option"+"Control"+"Shift"+U,  notice that you can get to the Main area in the Landmarks Menu. - only when. you have VoiceOver on.

# Affected Projects or Products
- JumpstartUI
- There should be no change visible on the UI unless you are using `tab` key to navigate.

# Associated Issues and/or People
- D8CORE-2444
- A11y
- @cjwest 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: lib/js/jumpstart_ui.behavior.js [hunk 1]
```diff
@@ -15,11 +15,20 @@ export default {
...
       });
+
+      // Changing the skip link from an anchor to a div.
+      const skiplinktarget = $('a#main-content');
+      const skiplinktargetContent = $(skiplinktarget).text();
+
+      skiplinktarget.wrap('<div id="main-content"></div>');
+      $('div#main-content').attr('tabindex', '-1')
+        .addClass('visually-hidden focusable').append(skiplinktargetContent);
+      skiplinktarget.remove();
```
Comment (pookmish) [inline lib/js/jumpstart_ui.behavior.js]: All of this js work can actually be done in PHP. Take a look to see if you can make it work in that direction. It'll give you a good way to learn more about render arrays.
see the [block code](https://github.com/SU-SWS/jumpstart_ui/blob/8.x-1.x/src/Plugin/Block/SkipNavAnchorBlock.php)
see the [html tag documentation](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21HtmlTag.php/class/HtmlTag/8.2.x)

Let me know if you aren't sure about it.

**Review by pookmish — CHANGES_REQUESTED** [review]

File: lib/js/jumpstart_ui.behavior.js (hunk 1 above)
Comment (jenbreese) [reply lib/js/jumpstart_ui.behavior.js]: Thank you @pookmish. That was exactly what I wanted to know. I think it belongs in our JumpstartUI, right?

Comment (pookmish) [reply lib/js/jumpstart_ui.behavior.js]: Yep.

Comment (jenbreese) [reply lib/js/jumpstart_ui.behavior.js]: @pookmish I got this to work with the php. 👍 I want to review with @cjwest. So I will move it to not ready for merging. Would that be the best solution?

File: src/Plugin/Block/SkipNavAnchorBlock.php [hunk 2]
```diff
@@ -24,8 +24,9 @@ public function build() {
     return [
       'anchor' => [
         '#title' => $this->t('Main Content'),
-        '#type' => 'link',
-        '#url' => Url::fromUri('internal:#' . $id),
+        '#type' => 'html_tag',
+        '#tag' => 'div',
+        '#value' => $this->t('Main Content'),
```
Comment (pookmish) [inline src/Plugin/Block/SkipNavAnchorBlock.php]: since you have the `#value`, the `#title` can be removed.

**Review by pookmish — CHANGES_REQUESTED** [review]
Just the 1 line cleanup and this is g2g for me

File: src/Plugin/Block/SkipNavAnchorBlock.php (hunk 2 above)
Comment (pookmish) [reply src/Plugin/Block/SkipNavAnchorBlock.php]: oh and the CodeClimate fix & tests

Comment (jenbreese) [reply src/Plugin/Block/SkipNavAnchorBlock.php]: Thanks! will fix.

File: tests/codeception/acceptance/ComponentsCest.php [hunk 3]
```diff
@@ -279,4 +279,15 @@ public function testQuote(AcceptanceTester $I) {
     $I->canSee('alt', '#content');
   }
 
+  /**
+   * Skiplink Component.
+   */
+  public function testSkiplink(AcceptanceTester $I) {
```
Comment (pookmish) [inline tests/codeception/acceptance/ComponentsCest.php]: You probably don't need a new test, just need to alter the existing test. https://github.com/SU-SWS/jumpstart_ui/blob/8.x-1.x/tests/src/Unit/Plugin/Block/SkipNavAnchorBlockTest.php#L81

**Comment by jenbreese** [comment]
@cjwest and I looked at my local and she liked it. We would need to ahve it merged so Clare and Jiatyan can try it out on dev.

**Review by pookmish — APPROVED** [review]

**Comment by cjwest** [comment]
👏

**Pull request merged**